src/test/kotlin/com/terraformation/backend/report/db/ReportStoreTest.kt

```diff
@@ -581,4 +595,202 @@ class ReportStoreTest : DatabaseTest(), RunsAsUser {
       assertEquals(listOf(missingReportOrganizationId), store.findOrganizationsForCreate())
     }
   }
+
+  @Nested
+  inner class FetchSettings {
+    @Test
+    fun `returns defaults if no settings have been saved`() {
+      assertEquals(
+          ReportSettingsModel(
+              isConfigured = false,
+              organizationId = organizationId,
+              organizationEnabled = true,
+              projects = emptyList()),
+          store.fetchSettingsByOrganization(organizationId),
+          "Result with no projects")
+
+      val projectId = insertProject()
+
+      assertEquals(
+          ReportSettingsModel(
+              isConfigured = false,
+              organizationId = organizationId,
+              organizationEnabled = true,
+              projects =
+                  listOf(
+                      ReportProjectSettingsModel(
+                          projectId = projectId, isConfigured = false, isEnabled = true))),
+          store.fetchSettingsByOrganization(organizationId),
+          "Result with unconfigured project")
+    }
+
+    @Test
+    fun `supports organization-level settings`() {
+      insertOrganizationReportSettings(isEnabled = false)
+
+      assertEquals(
+          ReportSettingsModel(
+              isConfigured = true,
+              organizationId = organizationId,
+              organizationEnabled = false,
+              projects = emptyList(),
+          ),
+          store.fetchSettingsByOrganization(organizationId),
+          "Organization reports should be disabled")
+
+      val otherOrganizationId = insertOrganization(2)
+      insertOrganizationReportSettings(organizationId = otherOrganizationId, isEnabled = true)
+
+      assertEquals(
+          ReportSettingsModel(
+              isConfigured = true,
+              organizationId = otherOrganizationId,
+              organizationEnabled = true,
+              projects = emptyList()),
+          store.fetchSettingsByOrganization(otherOrganizationId),
+          "Organization reports should be enabled")
+    }
+
+    @Test
+    fun `supports project-level settings`() {
+      insertOrganizationReportSettings(isEnabled = false)
+      val unconfiguredProjectId = insertProject()
+
+      assertEquals(
+          ReportSettingsModel(
+              isConfigured = true,
+              organizationId = organizationId,
+              organizationEnabled = false,
+              projects =
+                  listOf(
+                      ReportProjectSettingsModel(
+                          projectId = unconfiguredProjectId,
+                          isConfigured = false,
+                          isEnabled = true),
+                  )),
+          store.fetchSettingsByOrganization(organizationId),
+          "Projects list should be empty if no projects are enabled")
```

Comment:
The string description mentions empty projects list but the assert expects a non-empty projects list.

Reply:
Good catch. I was going back and forth on what the default return value should be. And in fact, this assertion isn't even useful at all because the default behavior is already covered by its own test.